pkg/generators/models/models.go

```diff
@@ -343,6 +343,7 @@ func fillValidationRelatedProperties(ref *openapi3.SchemaRef, spec *PropSpec) (i
 	importsMap := make(map[string]something)
 
 	if (len(spec.GoType) > 0 && spec.GoType[0] >= 'A' && spec.GoType[0] <= 'Z') ||
+		(len(spec.GoType) > 1 && spec.GoType[0] == '*' && spec.GoType[1] >= 'A' && spec.GoType[1] <= 'Z') ||
```

Comment:
Since I have not looked at the validation code before, I'm shocked to see this `if` statement now. This cannot be maintainable, we need a comment per each logical operation here.

Reply:
perhaps we can split this out into a function with a switch statement, ideally an annotated switch statement describing what kind of fields each branch is suppose to capture?

Reply:
I added comments to each line, I hope thats enough. This is already part of a small function (`fillValidationRelatedProperties`) so I don't see much value in extracting this further.

Reply:
I replaced the `if` with a documented regexp

Reply:
@LucasRoesler please have a final look.

Reply:
![Screenshot from 2021-04-28 15-36-15](https://user-images.githubusercontent.com/887952/116413021-843a1a00-a837-11eb-9a37-57ca6d0f5d65.png)